src/components/SkeletonLoader/SkeletonLoader.tsx

```diff
@@ -1,7 +1,119 @@
-import React from "react";
+import * as React from "react";
+import bem from "../../utils/bem";
+import {
+  SkeletonLoaderImageRatios,
+  SkeletonLoaderLayouts,
+} from "./SkeletonLoaderTypes";
 
-function SkeletonLoader() {
-  return <div className="skeleton-loader"></div>;
+export interface SkeletonLoaderProps {
+  /** Additional attributes passed to the horizontal rule */
+  attributes?: { [key: string]: any };
+  /** Optional boolean value to control visibility of border around skeleton loader */
+  border?: boolean;
+  /** ClassName you can add in addition to `skeleton-loader` */
+  className?: string;
+  /** Optional numeric value to control the number of lines for content placeholder; default value is `3` */
+  contentSize?: number;
+  /** Optional numeric value to control the number of lines for heading placeholder; default value is `1` */
+  headingSize?: number;
+  /** Optional value to control the aspect ratio of the image placeholder; default value is `square` */
+  imageAspectRatio?: SkeletonLoaderImageRatios;
+  /** Optional value to control the position of the image placeholder; default value is `vertical` */
+  layout?: SkeletonLoaderLayouts;
+  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
+  modifiers?: string[];
+  /** Optional boolean value to control visibility of button placeholder */
+  showButton?: boolean;
+  /** Optional boolean value to control visibility of content placeholder */
+  showContent?: boolean;
+  /** Optional boolean value to control visibility of image placeholder */
+  showImage?: boolean;
+  /** Optional boolean value to control visibility of heading placeholder */
+  showHeading?: boolean;
+  /** Optional width value.  This value should be entered with the same formatting as a CSS width attribute (ex. `50%`, `640px`, `20rem`).  If omitted, the horizobntal rule will have a default width of 100%. */
+  width?: string;
 }
 
-export default SkeletonLoader;
+export default function SkeletonLoader(
+  props: React.PropsWithChildren<SkeletonLoaderProps>
+) {
+  const {
+    attributes = {},
+    border,
+    className,
+    contentSize = 3,
+    headingSize = 1,
+    imageAspectRatio = SkeletonLoaderImageRatios.Square,
+    layout = SkeletonLoaderLayouts.Vertical,
+    showButton = false,
+    showContent = true,
+    showImage = true,
+    showHeading = true,
+    width = "auto",
+    modifiers = [],
+  } = props;
+
+  const imageModifiers = [];
+
+  attributes["style"] = {
+    width: width,
+  };
+
+  imageModifiers.push(imageAspectRatio);
```

Comment:
No other value gets added to the `imageModifiers` array. You can add the value in the declaration:
`const imageModifiers = [imageAspectRatio];`

Reply:
Thanks.  Done.